builder/azure/arm/step_get_ip_address.go

```diff
@@ -46,6 +48,8 @@ func NewStepGetIPAddress(client *AzureClient, ui packer.Ui, endpoint EndpointTyp
 		step.get = step.getPrivateIP
 	case PublicEndpoint:
 		step.get = step.getPublicIP
+	case PublicPrivateEndpoint:
```

Comment:
This name is confusing IMO, but I can see why you choose it.  Do you have any other ideas?  The only thing that I can think of VnetPublicIp, or make the code realize that it should select the public IP instead of the private IP.

I'll take a deeper look, and see if I can think of anything.